Framework/API/src/ExperimentInfo.cpp

```diff
@@ -302,31 +304,17 @@ void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
   }
   const auto parInstrument = Geometry::ParComponentFactory::createInstrument(
       sptr_instrument, m_parmap);
+
+  m_infoVisitor = makeOrRetrieveVisitor(*sptr_instrument);
```

Comment:
@SimonHeybrock Interesting... yes I agree. I think to get the full benefit from this we would need to have the `InfoComponentVisitor` capable of fully constructing `DetectorInfo`, otherwise we are back to exposing all the methods via `Instrument` to get/set the information for the `DetectorInfo`. So basically, I think this is a good idea, but we can't do this properly now.